sdk/sdk-core/aar/src/main/java/mil/emp3/core/mapgridlines/coordinates/UTMCoordinate.java

```diff
@@ -0,0 +1,726 @@
+     * @return String or null if the current one is "C".
+     */
+    public String getPreviousLetter() {
+        int iIndex = latBands.indexOf(zoneLetterValue);
+
+        if (zoneLetterValue.charAt(0) == 'C') {
+            return null;
+        }
+        iIndex--;
+
+        return latBands.charAt(iIndex) + "";
+    }
+
+
+    /**
+     * This method retrieves the zone letter after the current one.
+     * @return String or null if the current one is "X".
+     */
+    public String getNextLetter() {
+        int iIndex = latBands.indexOf(zoneLetterValue);
+
+        if (zoneLetterValue.charAt(0) == 'X') {
+            return null;
+        }
+        iIndex++;
+
+        return latBands.charAt(iIndex) + "";
+    }
+
+    /**
+     * This method retrieves the min northing value for the grid zone.
+     * @return
+     */
+    public double getMinNorthingForZone() {
+        return getMinNorthingForZone(this.zoneLetterValue);
+    }
+
+    /**
+     * This static method retrieves the min northing value for the zone letter specified.
+     * @param letter The zone letter
+     * @return double
+     */
+    public static double getMinNorthingForZone(String letter) {
+        double northing;
+
+        switch (letter) {
+            case "C":
+                northing = 1100000.0;
+                break;
+            case "D":
+                northing = 2000000.0;
+                break;
+            case "E":
+                northing = 2800000.0;
+                break;
+            case "F":
+                northing = 3700000.0;
+                break;
+            case "G":
+                northing = 4600000.0;
+                break;
+            case "H":
+                northing = 5500000.0;
+                break;
+            case "J":
+                northing = 6400000.0;
+                break;
+            case "K":
+                northing = 7300000.0;
+                break;
+            case "L":
+                northing = 8200000.0;
+                break;
+            case "M":
+                northing = 9100000.0;
+                break;
+            case "N":
+                northing = 0.0;
+                break;
+            case "P":
+                northing = 800000.0;
+                break;
+            case "Q":
+                northing = 1700000.0;
+                break;
+            case "R":
+                northing = 2600000.0;
+                break;
+            case "S":
+                northing = 3500000.0;
+                break;
+            case "T":
+                northing = 4400000.0;
+                break;
+            case "U":
+                northing = 5300000.0;
+                break;
+            case "V":
+                northing = 6200000.0;
+                break;
+            case "W":
+                northing = 7000000.0;
+                break;
+            case "X":
+                northing = 7900000.0;
+                break;
+            default:
```

Comment:
Might want to considwer converting this code into a table letter - "A" based table. It will make it efficient and readable. I think there are a couple of opportunities like this in other files also.